metrics/web/test/common/presentation/navigation/metrics_router_delegate_test.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:metrics/common/presentation/navigation/metrics_page/metrics_page.dart';
+import 'package:metrics/common/presentation/navigation/metrics_page/metrics_page_factory.dart';
+import 'package:metrics/common/presentation/navigation/metrics_router_delegate.dart';
+import 'package:metrics/common/presentation/navigation/route_configuration/route_name.dart';
+import 'package:metrics/common/presentation/navigation/state/navigation_notifier.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+import '../../../test_utils/matcher_util.dart';
+import '../../../test_utils/route_configuration_stub.dart';
+
+void main() {
+  group("MetricsRouterDelegate", () {
+    final navigationNotifierMock = _NavigationNotifierMock();
+    const configuration = RouteConfigurationStub(name: RouteName.dashboard);
+    final metricsRouterDelegate = MetricsRouterDelegate(navigationNotifierMock);
+
+    tearDown(() {
+      reset(navigationNotifierMock);
+    });
+
+    test(
+      "throws an AssertionError if the given navigation notifier is null",
+      () {
+        expect(
+          () => MetricsRouterDelegate(null),
+          MatcherUtil.throwsAssertionError,
+        );
+      },
+    );
+
+    test(
+      "throws an AssertionError if the given navigation observers are null",
+      () {
+        expect(
+          () => MetricsRouterDelegate(
+            navigationNotifierMock,
+            navigatorObservers: null,
+          ),
+          MatcherUtil.throwsAssertionError,
+        );
+      },
+    );
+
+    test(
+      "applies an empty list to the navigator observers if the parameter is not specified",
+      () {
+        final metricsRouterDelegate = MetricsRouterDelegate(
+          navigationNotifierMock,
+        );
+
+        expect(metricsRouterDelegate.navigatorObservers, equals([]));
+      },
+    );
+
+    test(
+      ".currentConfiguration provides the current route configuration of the navigation notifier",
+      () {
+        when(navigationNotifierMock.currentConfiguration).thenReturn(
+          configuration,
+        );
+
+        expect(
+          metricsRouterDelegate.currentConfiguration,
+          equals(configuration),
+        );
+      },
+    );
+
+    test(
+      ".navigatorKey provides the global object key with a value equals to the navigator notifier",
+      () {
+        final key = metricsRouterDelegate.navigatorKey as GlobalObjectKey;
+
+        expect(key.value, isA<NavigationNotifier>());
```

Comment:
```suggestion
        final navigationNotifier = NavigationNotifier(MetricsPageFactory());
        final expectedKey = GlobalObjectKey<NavigatorState>(navigationNotifier);

        final routerDelegate = MetricsRouterDelegate(navigationNotifier);
        final actualKey = routerDelegate.navigatorKey;

        expect(actualKey, equals(expectedKey));
```
What do you think about this?